### PR TITLE
Data: Use proper temp file in TestMetricsRowGroupFilterTypes

### PR DESCRIPTION
**Pull request by jzhuge**
This fixes TestMetricsRowGroupFilterTypes failures when 2 Jenkins jobs running this test simultaneously on the same Jenkins node. They create a race condition on hard-coded temp file "/tmp/stats-row-group-filter-types-test.parquet". Test error message is:

```
org.apache.iceberg.exceptions.NotFoundException: Failed to read file: /tmp/stats-row-group-filter-types-test.parquet
```

### CODE REVIEW

File: data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java [hunk 1]
```diff
@@ -135,8 +134,17 @@
       DateTimeFormatter.ISO_LOCAL_DATE_TIME);
   private static final byte[] fixed = "abcd".getBytes(StandardCharsets.UTF_8);
 
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  private File orcFile;
+  private File parquetFile;
```
Comment (kbendick) [inline data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java:141]: Do we need two separate variables for `orcFile` and `parquetFile`?

It seems like the tests only run on one or the other at a time.

But also, I understand if you don't want to have to reorganize the tests too much. 🙂

**Review by kbendick — APPROVED** [review]
A minor comment, but thanks for this fix @jzhuge!

I can see how the hardcoded files in `/tmp` could easily cause problems in some scenarios.

File: data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java [hunk 2]
```diff
@@ -173,48 +181,36 @@ public void createInputFile() throws IOException {
   }
 
   public void createOrcInputFile(List<Record> records) throws IOException {
-    if (ORC_FILE.exists()) {
-      Assert.assertTrue(ORC_FILE.delete());
-    }
```
Comment (kbendick) [inline data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java:178]: From my experience with this, the file won't write if it already exists.

So you can either create a directory and then create a file underneath to write to, or to keep things simple and just keep the present exists / delete (which will still ensure that the file clean up happens properly).

File: data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java (hunk 1 above)
Comment (rdblue) [reply data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java:141]: +1

We can probably keep a FileFormat and a File instead.

**Pull request closed without merging**